test/specs/palindrom.test.js

```diff
@@ -1,256 +1,210 @@
+        });
     });
-  });
 });
-describe("Palindrom", () => {
-  describe("#patching", () => {
-    beforeEach(() => {
-      moxios.install();
-    });
-    afterEach(() => {
-      moxios.uninstall();
-    });
-    it("should patch changes", function(done) {
-      moxios.stubRequest("http://localhost/testURL", {
-        status: 200,
-        headers: { contentType: "application/json" },
-        responseText: '{"hello": "world"}'
-      });
+describe('Palindrom', () => {
+    describe('#patching', () => {
+        beforeEach(() => {
+            
+        });
+        afterEach(() => {
+            fetchMock.restore();
+        });
+        it('should patch changes', async () => {
+            fetchMock.mock(getTestURL('testURL'), {
+                status: 200,
+                headers: { contentType: 'application/json' },
+                body: '{"hello": "world"}'
+            });
+
+            let tempObject;
+
+            new Palindrom({
+                remoteUrl: getTestURL('testURL'),
+                onStateReset: obj => (tempObject = obj)
+            });
 
-      const palindrom = new Palindrom({
-        remoteUrl: "http://localhost/testURL",
-        onStateReset: function(tempObject) {
-          assert.equal(tempObject.hello, "world");
-          tempObject.hello = "galaxy";
+            await sleep();
+            assert.equal(tempObject.hello, 'world');
+            tempObject.hello = 'galaxy';
 
-          /* now two ajax requests should had happened,
+            /* now two ajax requests should had happened,
                     the initial one, and the patch one (hello = world => hello = galaxy)*/
-          setTimeout(
-            () => {
-              assert.equal(2, moxios.requests.count());
-              let request = moxios.requests.mostRecent();
+            await sleep();
+            assert.equal(2, fetchMock.calls().length);
+            let request = fetchMock.lastOptions();
 
-              assert.equal(
+            assert.equal(
                 '[{"op":"replace","path":"/hello","value":"galaxy"}]',
-                request.config.data
-              );
-              done();
-            },
-            5
-          );
-        }
-      });
-    });
-    it("should not patch changes after unobserve() was called", function(done) {
-      moxios.stubRequest("http://localhost/testURL", {
-        status: 200,
-        headers: { contentType: "application/json" },
-        responseText: '{"unwatched": "object"}'
-      });
-      let tempObject;
-      const palindrom = new Palindrom({
-        remoteUrl: "http://localhost/testURL",
-        onStateReset: function(obj) {
-          tempObject = obj;
-        }
-      });
-      setTimeout(
-        () => {
-          assert.equal(1, moxios.requests.count());
-          assert.equal(tempObject.unwatched, "object");
-          tempObject.unwatched = "objecto";
-        },
-        5
-      );
+                JSON.parse(request.body)
```

Comment:
Why should we parse a strong to be compared to a string? If it matches it means that they are not equal! but `request.body` is a string with stringified object.

Reply:
Addressed, but would appreciate @warpech review

Reply:
It looks good now